pkg/machine/qemu/stubber.go

```diff
@@ -168,6 +166,28 @@ func (q *QEMUStubber) StartVM(mc *vmconfigs.MachineConfig) (func() error, func()
 	defer dnr.Close()
 	defer dnw.Close()
 
+	if q.mountType == MountTypeVirtiofs {
+		runtime, err := mc.RuntimeDir()
+		if err != nil {
+			return nil, nil, err
+		}
+		spawner, err := newVirtiofsdSpawner(runtime)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for _, hostmnt := range mc.Mounts {
+			qemuArgs, virtiofsdHelper, err := spawner.spawnForMount(hostmnt)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to init virtiofsd for mount %s: %w", hostmnt.Source, err)
+			}
+			q.Command = append(q.Command, qemuArgs...)
+			q.virtiofsHelpers = append(q.virtiofsHelpers, *virtiofsdHelper)
+		}
+	} else {
+		panic("Unhandled mount type:" + q.mountType)
```

Comment:
In Rust this would have started out as
```
enum Virtfs {
  NineP,
  Virtiofs
}
```

And this bit of code would have been a compiler-checked exhaustive `match` that went down to one element.  But there's no such thing in Go...changing this to return an error would be weird since it should be really unreachable code, rihgt?

Probably what I can do though is just eliminate the mountType variable in the last patch that drops 9p.